cvat/apps/consensus/merge_consensus_jobs.py

```diff
@@ -0,0 +1,102 @@
+from typing import List, Dict
+import datumaro as dm
+import django_rq
+from django.conf import settings
+from datumaro.components.operations import IntersectMerge
+from cvat.apps.quality_control.quality_reports import JobDataProvider
+from cvat.apps.engine.models import Job, JobType
+from cvat.apps.dataset_manager.bindings import import_dm_annotations
+from rest_framework import status
+from rest_framework.response import Response
+from cvat.apps.dataset_manager.task import patch_job_data, PatchAction
+from cvat.apps.engine.utils import get_rq_lock_by_user, get_rq_job_meta, define_dependent_job, process_failed_job
+from cvat.apps.engine.serializers import RqIdSerializer
+from django.utils import timezone
+from django.db import transaction
+
+def get_consensus_jobs(task_id: int):
+    jobs = {} # parent_job_id -> [consensus_job_id]
+    for job in Job.objects.select_related("segment").filter(segment__task_id=task_id, type=JobType.ANNOTATION.value).order_by('id'):
+        if job.parent_job_id is not None:
+            if job.parent_job_id not in jobs:
+                jobs[job.parent_job_id] = []
+            jobs[job.parent_job_id].append(job.id)
+    return jobs
+
+def get_annotations(job_id: int):
+    return JobDataProvider(job_id).dm_dataset
+
+
+@transaction.atomic
+def _merge_consensus_jobs(task_id: int):
+    jobs = get_consensus_jobs(task_id)
+    merger = IntersectMerge()
+
+    for parent_job_id, job_ids in jobs.items():
+        consensus_dataset = list(map(get_annotations, job_ids))
+
+        merged_dataset: dm.Dataset = merger(consensus_dataset)
+
+        # check if the merged dataset has annotations
+        for item in merged_dataset:
+            if not item.annotations:
+                return 400
```

Comment:
This check doesn't seem to be correct, as some images can have nothing to be annotated. I'd recommend just removing it, as it seems to be some debugging code for development. It should be covered by a unit test later.